tests/t122-vector.c

```diff
@@ -23,6 +23,12 @@ int main(int argc, char **argv) {
   CeedVectorSetArray(y, CEED_MEM_HOST, CEED_COPY_VALUES, a);
 
   // Test multiplying two vectors into third
+  {
+    // Sync memtype to device for GPU backends
+    CeedMemType type = CEED_MEM_HOST;
+    CeedGetPreferredMemType(ceed, &type);
+    CeedVectorSyncArray(w, type);
```

Comment:
I don't think these are actually needed -- this test already ran on the GPU before.  Since `w` (and `y`) isn't set to anything yet, it isn't set to `CEED_*_HOST_SYNC`, the output vec's `memState` is the one used to determine whether to call the host or device function. 
Not that it hurts anything to have it here, though.

Reply:
Good point. I'm thinking of leaving it there it make it clear that this occurs in the backend's preferred memory space.